### PR TITLE
Bump the minimum required version of Node.js to 20.x

### PR DESCRIPTION
**Pull request by desrosj**
Trac ticket: https://core.trac.wordpress.org/ticket/59663

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by desrosj** [comment]
I've made a [request to the systems team to install 20.x on the build server](https://make.wordpress.org/systems/2023/11/27/install-node-js-20-x-on-the-build-server/). It's possible we can skip 18.x all together, though I'd advise we proceed as though we'll only have 18.x available for this current cycle (WP 6.5). We can always bump a second time should something change.

**Comment by desrosj** [comment]
@swissspidy It seems that this change will cause the comparison between the previous branch and this one to fail because of the differing versions of Node.js/npm between them. Any idea around the best way to resolve this?

I do plan to backport this update to the 6.4 branch because that is currently the only other one that uses `16.x` (all previous ones use `14.x`). 

Will that then cause the workflow to fail in the 6.4 branch when comparing the 6.3 branch? I'd expect it to be failing currently since the 6.3 branch uses 14.x and not 16.x. 🤔

**Comment by joemcgill** [comment]
@desrosj I think the old ones aren't failing because this will only happen when the workflow is run when comparing the current commit to the previous commit when the previous commit uses a different node version. Perhaps adding an `nvm use` or similar command before running `npm ci` when comparing against the previous commit will suffice? At any rate, I would expect the very next commit to resolve the issue.

**Comment by swissspidy** [comment]
What Joe says :) `nvm install` after "Check out target commit" and then again after "Reset to original commit" should do the trick. Should be safe to add, even if it theoretically only happens once when doing the version bump. For future commits it's a no-op because the Node version doesn't change between commits. So a reasonable addition 👍

File: .github/workflows/performance.yml [hunk 1]
```diff
@@ -188,8 +189,11 @@ jobs:
           fi
           git reset --hard $TARGET_SHA
 
-      - name: Check desired node version
-        run: nvm install
+      - name: Set up Node.js
```
Comment (swissspidy) [inline .github/workflows/performance.yml:195]: We later call `git reset --hard $GITHUB_SHA` in the `Reset to original commit` step. Before that, the same is probably needed.

Comment (desrosj) [reply .github/workflows/performance.yml:195]: Should be resolved now!

**Comment by desrosj** [comment]
Merged in https://core.trac.wordpress.org/changeset/57212 and https://core.trac.wordpress.org/changeset/57213.

**Pull request closed without merging**